### ISSUE TITLE
Memory leak if transition aborted before animateOut/destroy called

### ISSUE THREAD

**Issue by trompx** [bigwheel-framework/bigwheel#40]
Hey,

Let's take as an example the transitions between two pages: home -> about -> home.
If I trigger the about -> home before the first home has animated out and destroyed, the animatOut and destroy function of the first instance of home are never called, resulting in duplicate home page in the DOM (and memory leak : see in chrome performance devtools the number of DOM elements keep growing).
I guess a router module in the bigwheel framework hold the pages instances objects, but I don't know how it manage their data.

As a workaround, would it be possible to create a global variable holding the pages object and in the case of an aborted animateOut, being able on the next animateIn of the same page to force animateOut / destroy of the old instance?

**Comment by mikkoh** [comment]
Hmmm. Aborting animations isn't something I've ever considered. Why would you want to do this? To me if you want to handle aborting an animate out there likely needs to be more logic added to bigwheel.

**Comment by trompx** [comment]
Sorry I completely missed your answer.
The usecase was when a user goes back with the browser back button, say multiple times, as animations don't have time to finish, the memory leak I described occurs.

If I remember correctly, I solved this by saving page instances and on each animate out of each page, loop all page instances and manually destroy the one (of the same page type) who did not finish animating out.
